packages/react-components/src/trace-explorer/trace-explorer-time-range-data-widget.tsx

```diff
@@ -0,0 +1,353 @@
+import { TimeRange } from 'traceviewer-base/lib/utils/time-range';
+
+export interface ReactTimeRangeDataWidgetProps {
+    id: string,
+    title: string,
+}
+
+export interface ReactTimeRangeDataWidgetState {
+    activeExperiment?: string,
+    activeData?: ExperimentTimeRangeData,
+    userInputSelectionStartIsValid: boolean,
+    userInputSelectionEndIsValid: boolean,
+    userInputSelectionStart?: bigint,
+    userInputSelectionEnd?: bigint,
+    inputting: boolean,
+}
+
+export interface ExperimentTimeRangeData {
+    experimentId: string;
+    // 99% of the time all of the values below will be defined.
+    // We need to make them possibly undefined because they
+    // come in at different times from different signals when loading.
+    name?: string;
+    absoluteRange?: { start: bigint, end: bigint };
+    viewRange?: { start: bigint, end: bigint };
+    selectionRange?: { start: bigint, end: bigint };
+}
+
+export class ReactTimeRangeDataWidget extends React.Component<ReactTimeRangeDataWidgetProps, ReactTimeRangeDataWidgetState> {
+
+    private selectionStartInput: React.RefObject<HTMLInputElement>;
+    private selectionEndInput: React.RefObject<HTMLInputElement>;
+
+    private experimentDataMap: Map<string, ExperimentTimeRangeData>;
+
+    constructor(props: ReactTimeRangeDataWidgetProps) {
+        super(props);
+        this.selectionEndInput = React.createRef();
+        this.selectionStartInput = React.createRef();
+        this.experimentDataMap = new Map<string, ExperimentTimeRangeData>();
+        this.state = {
+            inputting: false,
+            userInputSelectionStartIsValid: true,
+            userInputSelectionEndIsValid: true,
+        };
+        this.subscribeToEvents();
+    }
+
+    private subscribeToEvents = (): void => {
+        signalManager().on(Signals.VIEW_RANGE_UPDATED, this.onViewRangeUpdated);
+        signalManager().on(Signals.SELECTION_RANGE_UPDATED, this.onSelectionRangeUpdated);
+        signalManager().on(Signals.TRACEVIEWERTAB_ACTIVATED, this.onActiveExperimentUpdated);
+        signalManager().on(Signals.EXPERIMENT_SELECTED, this.onActiveExperimentUpdated);
+        signalManager().on(Signals.TRACE_OPENED, this.onActiveExperimentUpdated);
+        signalManager().on(Signals.EXPERIMENT_UPDATED, this.onAbsoluteRangeUpdate);
+        signalManager().on(Signals.EXPERIMENT_CLOSED, this.onExperimentClosed);
+    };
+
+    public componentWillUnmount = (): void => {
+        signalManager().off(Signals.VIEW_RANGE_UPDATED, this.onViewRangeUpdated);
+        signalManager().off(Signals.SELECTION_RANGE_UPDATED, this.onSelectionRangeUpdated);
+        signalManager().off(Signals.TRACEVIEWERTAB_ACTIVATED, this.onActiveExperimentUpdated);
+        signalManager().off(Signals.EXPERIMENT_SELECTED, this.onActiveExperimentUpdated);
+        signalManager().off(Signals.TRACE_OPENED, this.onActiveExperimentUpdated);
+        signalManager().off(Signals.EXPERIMENT_UPDATED, this.onAbsoluteRangeUpdate);
```

Comment:
shouldn't the method be called `onExperimentUpdated` instead of `onAbsoluteRangeUpdate`?